spec/lib/whitehall_importer/create_edition_spec.rb

```diff
@@ -41,67 +39,231 @@
+                             whitehall_edition: whitehall_edition,
+                             edition_number: 1,
+                             user_ids: user_ids)
+      end
+
+      it "creates two statuses" do
         expect(edition.statuses.count).to eq(2)
+      end
+
+      it "sets the correct withdrawn status" do
+        expect(edition).to be_withdrawn
+      end
+
+      it "sets the correct previous status" do
         expect(edition.statuses.first).to be_published
       end
+
+      it "creates a withdrawal record" do
+        expect(edition.status.details).to be_an_instance_of(Withdrawal)
+      end
+
+      it "sets the correct withdrawal metadata" do
+        expect(edition.status.details.withdrawn_at.rfc3339).to eq(
+          whitehall_edition["unpublishing"]["created_at"],
+        )
+        expect(edition.status.details.public_explanation).to eq(
+          whitehall_edition["unpublishing"]["explanation"],
+        )
+      end
+    end
+
+    context "when an unpublished edition has not been edited" do
+      let(:whitehall_edition) do
+        build(:whitehall_export_edition,
+              revision_history: [
+                { "event" => "create", "state" => "draft", "whodunnit" => 1, "created_at" => "2017-11-24T15:51:23+00:00" },
+                { "event" => "update", "state" => "published", "whodunnit" => 1, "created_at" => "2018-11-25T15:51:23+00:00" },
+                { "event" => "update", "state" => "draft", "whodunnit" => 1, "created_at" => "2019-11-26T15:51:23+00:00" },
+              ],
+              unpublishing: build(:whitehall_export_unpublishing,
+                                  alternative_url: "https://www.gov.uk/flextension",
+                                  unpublishing_reason: "Consolidated into another GOV.UK page",
+                                  explanation: "Brexit is being delayed again"))
+      end
+      let(:edition) do
+        described_class.call(document: document,
+                             current: true,
+                             whitehall_edition: whitehall_edition,
+                             edition_number: 1,
+                             user_ids: user_ids)
+      end
+
+      it "creates an edition with a status of removed" do
+        expect(edition.removed?).to be_truthy
+      end
+
+      it "creates a removal record" do
+        expect(edition.status.details).to be_an_instance_of(Removal)
+      end
+
+      it "sets the correct removal metadata" do
+        removal = edition.status.details
+        expect(removal.explanatory_note).to eq(whitehall_edition["unpublishing"]["explanation"])
+        expect(removal.alternative_path).to eq(whitehall_edition["unpublishing"]["alternative_url"])
+        expect(removal.redirect).to be_truthy
+      end
+
+      it "sets the correct timestamps on the edition" do
+        expect(edition.created_at).to eq("2017-11-24T15:51:23+00:00")
+        expect(edition.updated_at).to eq("2019-11-26T15:51:23+00:00")
```

Comment:
should we use let variables for this since this is quite removed from the setup?

Reply:
Could potentially check the timestamps in the check for a removed edition?

Reply:
Variables are now being used! 🌵